coverage/exceptions.py

```diff
@@ -24,11 +34,61 @@ class DataError(CoverageException):
     """An error in using a data file."""
     pass
 
+
 class NoDataError(CoverageException):
     """We didn't have data to work with."""
     pass
 
 
+class DataFileOrDirectoryNotFoundError(NoDataError):
+    """A data file or data directory could be found."""
+    @classmethod
+    def new_for_data_file_or_directory(
+        cls, data_file_or_directory_path: str, *, is_combining: bool = False
+    ) -> 'DataFileOrDirectoryNotFoundError':
+        """
+        Create a new instance.
+        """
+        message = (
+            f"The data file or directory `{os.path.abspath(data_file_or_directory_path)}` could not"
```

Comment:
You've used backticks here and elsewhere, as if these are Markdown, but they are not.  Better to use simple single quotes.